### PR TITLE
added jax.numpy.pad function

### PR DESCRIPTION
**Pull request by x0axz**
Close https://github.com/unifyai/ivy/issues/11779

### CODE REVIEW

**Comment by x0axz** [comment]
@Infrared1029 Please review this PR.

**Comment by x0axz** [comment]
@Infrared1029 Please review this PR.

**Comment by Infrared1029** [comment]
Hi @x0axz, the tests seem to fail, can you make sure the tests are passing so I can take a close look? Thanks!

**Comment by x0axz** [comment]
@Infrared1029 Thanks for pointing this out. I fixed all the tests and tested them locally. Please review it now.

**Comment by x0axz** [comment]
@Infrared1029 Thanks for reviewing. I fixed the failed tests. Please review now.

**Comment by Infrared1029** [comment]
hey @x0axz, even though the tests appear to be passing, the function does not seem to be implemented properly, which means the test is probably not covering the entire input space of the function, a simple falsifying example is the following:
```python
import ivy 
import jax.numpy as jnp
import ivy.functional.frontends.jax.numpy as ivy_jnp

jnp.pad(jnp.array([[1,2,3]]), ((1,1), (2,2))).shape # returns (3, 7)

ivy_jnp.pad(ivy_jnp.array([[1,2,3]]), ((1,1), (2,2))).shape # errors out
```

make sure the test indeed covers the entire input space and the function is implemented correctly.

**Comment by x0axz** [comment]
@Infrared1029 Hi, I've tried to make changes and been through the [documentation](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.pad.html#jax.numpy.pad) multiple times, but this implementation is according to the documentation. Will you please explain what you mean by "the function does not seem to be implemented properly" and "the test is probably not covering the entire input space of the function"?

**Comment by Infrared1029** [comment]
Hello @x0axz, as mentioned in my previous comment this code example fails 

```python
import ivy.functional.frontends.jax.numpy as ivy_jnp
ivy_jnp.pad(ivy_jnp.array([[1,2,3]]), ((1,1), (2,2))) # errors out
```
this should not fail if everything is implemented correctly, the fact that the tests are passing despite an example like this failing means the tests are not generating all possible inputs the function could take (because if that was the case, the test would not have passed)

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.
